posydon/unit_tests/popsyn/test_norm_pop.py

```diff
@@ -0,0 +1,701 @@
+import numpy as np
+import os
+import pandas as pd
+import pytest
+from pytest import fixture, mark
+
+import posydon.popsyn.IMFs as IMFs
+from posydon.popsyn.independent_sample import generate_independent_samples
+from posydon.config import PATH_TO_POSYDON
+
+# file to test
+from posydon.popsyn import norm_pop
+
+# Dummy IMF to simulate a valid IMF pdf
+class DummyIMF:
+    def __init__(self, m_min, m_max):
+        self.m_min = m_min
+        self.m_max = m_max
+    def pdf(self, m1):
+        # returns a constant value for testing
+        return 0.5
+
+class TestGetIMFPdf:
+    def test_invalid_imf_returns_constant(self):
+        kwargs = {
+            'primary_mass_scheme': 'NonExistentIMF',
+            'primary_mass_min': 1,
+            'primary_mass_max': 100
+        }
+        pdf_func = norm_pop.get_IMF_pdf(kwargs)
+        m_test = np.array([1, 10, 50])
+        result = pdf_func(m_test)
+        assert np.all(result == 1)        
+    
+    def test_valid_imf_returns_dummy_pdf(self):
+        # Monkey-patch: add DummyIMF to IMFs temporarily
+        IMFs.DummyIMF = DummyIMF
+        kwargs = {
+            'primary_mass_scheme': 'DummyIMF',
+            'primary_mass_min': 1,
+            'primary_mass_max': 100
+        }
+        pdf_func = norm_pop.get_IMF_pdf(kwargs)
+        m_test = np.array([1, 10, 50])
+        result = pdf_func(m_test)
+        # Expect DummyIMF.pdf to return 0.5 regardless of m1
+        expected = 0.5
+        assert np.allclose(result, expected)
+        # cleanup monkey-patch
+        del IMFs.DummyIMF
+
+    def test_imf_with_scheme_kwargs(self):
+        # Create a test IMF class that uses scheme kwargs
+        class SchemeKwargsIMF:
+            def __init__(self, m_min, m_max, custom_param=None):
+                self.m_min = m_min
+                self.m_max = m_max
+                self.custom_param = custom_param
+            def pdf(self, m1):
+                # Return custom_param if set, otherwise 1
+                return self.custom_param if self.custom_param is not None else 1
+        
+        # Monkey-patch the IMF class into IMFs module
+        IMFs.SchemeKwargsIMF = SchemeKwargsIMF
+        
+        # Create kwargs with scheme-specific parameters
+        kwargs = {
+            'primary_mass_scheme': 'SchemeKwargsIMF',
+            'primary_mass_min': 1,
+            'primary_mass_max': 100,
+            'SchemeKwargsIMF': {'custom_param': 3.5}  # Scheme-specific kwargs
+        }
+        
+        # Get the PDF function
+        pdf_func = norm_pop.get_IMF_pdf(kwargs)
+        
+        # Test that the custom parameter was used
+        m_test = np.array([1, 10, 50])
+        result = pdf_func(m_test)
+        expected = 3.5  # The value of custom_param
```

Comment:
```suggestion
        expected = kwargs[kwargs['primary_mass_scheme']]['custom_param']
```
This would avoid hard coding the value.